diff --git a/net/dnscrypt-proxy2/Makefile b/net/dnscrypt-proxy2/Makefile
index 64032bc41..e31b9d3a3 100644
--- a/net/dnscrypt-proxy2/Makefile
+++ b/net/dnscrypt-proxy2/Makefile
@@ -38,6 +38,13 @@ define Package/dnscrypt-proxy2
   CONFLICTS:=dnscrypt-proxy
 endef
 
+define Build/Compile
+$(call GoPackage/Build/Compile,-ldflags "-s -w")
+	chmod +x $(GO_PKG_BUILD_BIN_DIR)/dnscrypt-proxy && $(STAGING_DIR_HOST)/bin/upx --lzma --best $(GO_PKG_BUILD_BIN_DIR)/dnscrypt-proxy
+	[ ! -f $(PKG_BUILD_DIR)/public-resolvers.md ] && wget https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v3/public-resolvers.md -O $(PKG_BUILD_DIR)/public-resolvers.md
+	[ ! -f $(PKG_BUILD_DIR)/public-resolvers.md.minisig ] && wget https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v3/public-resolvers.md.minisig -O $(PKG_BUILD_DIR)/public-resolvers.md.minisig
+endef
+
 define Package/dnscrypt-proxy2/install
 	$(call GoPackage/Package/Install/Bin,$(PKG_INSTALL_DIR))
 
@@ -47,9 +54,8 @@ define Package/dnscrypt-proxy2/install
 	$(INSTALL_DIR) $(1)/etc/dnscrypt-proxy2
 	$(INSTALL_CONF) $(PKG_BUILD_DIR)/dnscrypt-proxy/example-dnscrypt-proxy.toml $(1)/etc/dnscrypt-proxy2/dnscrypt-proxy.toml
 	$(INSTALL_CONF) ./files/blocked-names.txt $(1)/etc/dnscrypt-proxy2/blocked-names.txt
-
-	$(INSTALL_DIR) $(1)/etc/init.d
-	$(INSTALL_BIN) ./files/dnscrypt-proxy.init $(1)/etc/init.d/dnscrypt-proxy
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/public-resolvers.md $(1)/etc/dnscrypt-proxy2/public-resolvers.md
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/public-resolvers.md.minisig $(1)/etc/dnscrypt-proxy2/public-resolvers.md.minisig
 
 	sed -i "s/^listen_addresses = .*/listen_addresses = ['127.0.0.53:53']/" $(1)/etc/dnscrypt-proxy2/dnscrypt-proxy.toml
 	sed -i "s/^  # blocked_names_file = 'blocked-names.txt'/blocked_names_file = 'blocked-names.txt'/" $(1)/etc/dnscrypt-proxy2/dnscrypt-proxy.toml
diff --git a/net/udpspeeder/Makefile b/net/udpspeeder/Makefile
index 49d24a245..d37b56746 100644
--- a/net/udpspeeder/Makefile
+++ b/net/udpspeeder/Makefile
@@ -48,12 +48,6 @@ endef
 define Package/UDPspeeder/install
 	$(INSTALL_DIR) $(1)/usr/bin
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/speederv2_cross $(1)/usr/bin/udpspeeder
-
-	$(INSTALL_DIR) $(1)/etc/config
-	$(INSTALL_CONF) ./files/udpspeeder-config $(1)/etc/config/udpspeeder
-
-	$(INSTALL_DIR) $(1)/etc/init.d
-	$(INSTALL_BIN) ./files/udpspeeder-init $(1)/etc/init.d/udpspeeder
 endef
 
 $(eval $(call BuildPackage,UDPspeeder))
diff --git a/utils/coremark/Makefile b/utils/coremark/Makefile
index bf7394f10..d5bd2d6de 100644
--- a/utils/coremark/Makefile
+++ b/utils/coremark/Makefile
@@ -80,6 +80,17 @@ endef
 define Package/coremark/install
 	$(INSTALL_DIR) $(1)/bin
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/coremark $(1)/bin/
+	$(INSTALL_DIR) $(1)/etc
+	$(INSTALL_BIN) ./coremark.sh $(1)/etc/
+	$(INSTALL_DIR) $(1)/etc/uci-defaults
+	$(INSTALL_BIN) ./coremark $(1)/etc/uci-defaults/xxx-coremark
+endef
+
+define Package/coremark/postinst
+#!/bin/sh
+[ -n "$${IPKG_INSTROOT}" ] || sed -i '/coremark/d' /etc/crontabs/root
+[ -n "$${IPKG_INSTROOT}" ] || echo "0 4 * * * /etc/coremark.sh" >> /etc/crontabs/root
+[ -n "$${IPKG_INSTROOT}" ] || crontab /etc/crontabs/root
 endef
 
 $(eval $(call BuildPackage,coremark))
