diff -uNr a/dnscrypt-proxy/xtransport.go b/dnscrypt-proxy/xtransport.go
--- a/dnscrypt-proxy/xtransport.go	2020-03-11 23:12:47.232361800 +0800
+++ b/dnscrypt-proxy/xtransport.go	2020-03-11 23:18:43.329268300 +0800
@@ -14,6 +14,7 @@
 	"net"
 	"net/http"
 	"net/url"
+	"syscall"
 	"strconv"
 	"strings"
 	"sync"
@@ -28,6 +29,7 @@
 
 const (
 	DefaultFallbackResolver = "9.9.9.9:53"
+	TCP_FASTOPEN_CONNECT    = 30
 	DefaultKeepAlive        = 5 * time.Second
 	DefaultTimeout          = 30 * time.Second
 	SystemResolverIPTTL     = 24 * time.Hour
@@ -148,7 +150,17 @@
 			}
 			addrStr = ipOnly + ":" + strconv.Itoa(port)
 			if xTransport.proxyDialer == nil {
-				dialer := &net.Dialer{Timeout: timeout, KeepAlive: timeout, DualStack: true}
+				dialer := &net.Dialer{Timeout: timeout, KeepAlive: timeout, DualStack: true,
+					Control: func(network, address string, c syscall.RawConn) error {
+						return c.Control(func(fd uintptr) {
+							err := syscall.SetsockoptInt(int(fd), syscall.SOL_TCP, TCP_FASTOPEN_CONNECT, 1)
+							if err != nil {
+								dlog.Debug("failed to set TCP_FASTOPEN_CONNECT=1")
+								return
+							}
+						})
+					},
+				}
 				return dialer.DialContext(ctx, network, addrStr)
 			}
 			return (*xTransport.proxyDialer).Dial(network, addrStr)
