diff --git a/package/lean/rblibtorrent/Makefile b/package/lean/rblibtorrent/Makefile
index d2709c7bb5..8deccdf2c2 100644
--- a/package/lean/rblibtorrent/Makefile
+++ b/package/lean/rblibtorrent/Makefile
@@ -1,13 +1,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=rblibtorrent
-PKG_VERSION:=1.1.13
+PKG_VERSION:=1.1.14
 PKG_RELEASE=2
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL:=https://github.com/arvidn/libtorrent.git
 PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
-PKG_SOURCE_VERSION:=6f1250c6535730897909240ea0f4f2a81937d21a
+PKG_SOURCE_VERSION:=244f0f189ba8ab9801e7fcf553beebfb83d7c86b
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 
 PKG_LICENSE:=BSD
