diff --git a/net/dnscrypt-proxy2/Makefile b/net/dnscrypt-proxy2/Makefile
index 64032bc41..e31b9d3a3 100644
--- a/net/dnscrypt-proxy2/Makefile
+++ b/net/dnscrypt-proxy2/Makefile
@@ -38,6 +38,13 @@ define Package/dnscrypt-proxy2
   CONFLICTS:=dnscrypt-proxy
 endef
 
+define Build/Compile
+$(call GoPackage/Build/Compile,-ldflags "-s -w")
+	chmod +x $(GO_PKG_BUILD_BIN_DIR)/dnscrypt-proxy && $(STAGING_DIR_HOST)/bin/upx --lzma --best $(GO_PKG_BUILD_BIN_DIR)/dnscrypt-proxy
+	[ ! -f $(PKG_BUILD_DIR)/public-resolvers.md ] && wget https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v3/public-resolvers.md -O $(PKG_BUILD_DIR)/public-resolvers.md
+	[ ! -f $(PKG_BUILD_DIR)/public-resolvers.md.minisig ] && wget https://raw.githubusercontent.com/DNSCrypt/dnscrypt-resolvers/master/v3/public-resolvers.md.minisig -O $(PKG_BUILD_DIR)/public-resolvers.md.minisig
+endef
+
 define Package/dnscrypt-proxy2/install
 	$(call GoPackage/Package/Install/Bin,$(PKG_INSTALL_DIR))
 
@@ -47,9 +54,8 @@ define Package/dnscrypt-proxy2/install
 	$(INSTALL_DIR) $(1)/etc/dnscrypt-proxy2
 	$(INSTALL_CONF) $(PKG_BUILD_DIR)/dnscrypt-proxy/example-dnscrypt-proxy.toml $(1)/etc/dnscrypt-proxy2/dnscrypt-proxy.toml
 	$(INSTALL_CONF) ./files/blocked-names.txt $(1)/etc/dnscrypt-proxy2/blocked-names.txt
-
-	$(INSTALL_DIR) $(1)/etc/init.d
-	$(INSTALL_BIN) ./files/dnscrypt-proxy.init $(1)/etc/init.d/dnscrypt-proxy
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/public-resolvers.md $(1)/etc/dnscrypt-proxy2/public-resolvers.md
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/public-resolvers.md.minisig $(1)/etc/dnscrypt-proxy2/public-resolvers.md.minisig
 
 	sed -i "s/^listen_addresses = .*/listen_addresses = ['127.0.0.53:53']/" $(1)/etc/dnscrypt-proxy2/dnscrypt-proxy.toml
 	sed -i "s/^  # blocked_names_file = 'blocked-names.txt'/blocked_names_file = 'blocked-names.txt'/" $(1)/etc/dnscrypt-proxy2/dnscrypt-proxy.toml
diff --git a/net/softethervpn/Makefile b/net/softethervpn/Makefile
index 04450d198..b9c5d29b6 100644
--- a/net/softethervpn/Makefile
+++ b/net/softethervpn/Makefile
@@ -43,14 +43,24 @@ HOST_MAKE_FLAGS += -f src/makefiles/linux_$(if $(shell uname -m | grep 64),64,32
 HOST_LDFLAGS += -Wl,-rpath$(comma)$(STAGING_DIR_HOSTPKG)/lib
 # Prevent calling upstream configure
 define Host/Configure
+	$(SED) 's/^OPTIONS_COMPILE_RELEASE=/&-DUSE_INTEL_AESNI_LIBRARY /;' \
+		-e "s|^OPTIONS_LINK_RELEASE=-O2 |&$(TOPDIR)/../intel_aes_lib/intel_aes64.a |" \
+		$(HOST_BUILD_DIR)/src/makefiles/linux_64bit.mak
 endef
 
 define Host/Compile
 	# Build hamcorebuilder using host compiler and let it generate
 	# the hamcore.se2 archive file
 	# CFLAGS, CPPFLAGS & LDFLAGS need to be passed with CC because they are being ingored
+	(cd $(TOPDIR)/../intel_aes_lib; \
+	$(MAKE) clean; \
+	CC="$(HOSTCC) $(HOST_CFLAGS) $(HOST_CPPFLAGS) $(HOST_LDFLAGS)" $(MAKE); \
+	cd $(HOST_BUILD_DIR); \
 	CC="$(HOSTCC) $(HOST_CFLAGS) $(HOST_CPPFLAGS) $(HOST_LDFLAGS)" $(MAKE) $(HOST_MAKE_FLAGS) \
-		src/bin/BuiltHamcoreFiles/unix/hamcore.se2
+		src/bin/BuiltHamcoreFiles/unix/hamcore.se2; \
+	cd $(TOPDIR)/../intel_aes_lib; \
+	$(MAKE) clean; \
+	CC="$(TARGET_CC) $(TARGET_CFLAGS) $(TARGET_CPPFLAGS) $(TARGET_LDFLAGS)" $(MAKE))
 endef
 
 define Host/Install
@@ -73,6 +83,9 @@ MAKE_FLAGS += \
 	CC="$(TARGET_CC) $(TARGET_CFLAGS) $(TARGET_CPPFLAGS) $(TARGET_LDFLAGS)"
 
 define Build/Configure
+	$(SED) 's/^OPTIONS_COMPILE_RELEASE=/&-DUSE_INTEL_AESNI_LIBRARY /;' \
+		-e "s|^OPTIONS_LINK_RELEASE=-O2 |&$(TOPDIR)/../intel_aes_lib/intel_aes64.a |" \
+		$(PKG_BUILD_DIR)/src/makefiles/linux_64bit.mak
 	# Fetch prebuilt hamcore.se2 from staging dir
 	$(INSTALL_DIR) $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix
 	$(CP) $(HAMCORE_SE2) $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix/hamcore.se2
diff --git a/net/udpspeeder/Makefile b/net/udpspeeder/Makefile
index 49d24a245..d37b56746 100644
--- a/net/udpspeeder/Makefile
+++ b/net/udpspeeder/Makefile
@@ -48,12 +48,6 @@ endef
 define Package/UDPspeeder/install
 	$(INSTALL_DIR) $(1)/usr/bin
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/speederv2_cross $(1)/usr/bin/udpspeeder
-
-	$(INSTALL_DIR) $(1)/etc/config
-	$(INSTALL_CONF) ./files/udpspeeder-config $(1)/etc/config/udpspeeder
-
-	$(INSTALL_DIR) $(1)/etc/init.d
-	$(INSTALL_BIN) ./files/udpspeeder-init $(1)/etc/init.d/udpspeeder
 endef
 
 $(eval $(call BuildPackage,UDPspeeder))
diff --git a/utils/coremark/Makefile b/utils/coremark/Makefile
index bf7394f10..d5bd2d6de 100644
--- a/utils/coremark/Makefile
+++ b/utils/coremark/Makefile
@@ -80,6 +80,17 @@ endef
 define Package/coremark/install
 	$(INSTALL_DIR) $(1)/bin
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/coremark $(1)/bin/
+	$(INSTALL_DIR) $(1)/etc
+	$(INSTALL_BIN) ./coremark.sh $(1)/etc/
+	$(INSTALL_DIR) $(1)/etc/uci-defaults
+	$(INSTALL_BIN) ./coremark $(1)/etc/uci-defaults/xxx-coremark
+endef
+
+define Package/coremark/postinst
+#!/bin/sh
+[ -n "$${IPKG_INSTROOT}" ] || sed -i '/coremark/d' /etc/crontabs/root
+[ -n "$${IPKG_INSTROOT}" ] || echo "0 4 * * * /etc/coremark.sh" >> /etc/crontabs/root
+[ -n "$${IPKG_INSTROOT}" ] || crontab /etc/crontabs/root
 endef
 
 $(eval $(call BuildPackage,coremark))
